lib/swagger-oauth.js

```diff
@@ -165,6 +197,28 @@ function handleLogin() {
   return;
 }
 
+function handlePasswordFlow(auth) {
+  var tokenUri = auth.tokenUrl;
+
+  var authParams = {
+    grant_type: 'password',
+    client_id: encodeURIComponent(clientId),
```

Comment:
In my last push I added the client secret to the auth headers for basic auth.  I believe Spring Security requires basic auth for getting a token.  Does it solve your problem as well @pkerspe or does it need to be a query param?